src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -1,346 +1,332 @@
-     * @param {Boolean} isUnreadChat - Flag to check if the chat is unread in the LHN. Used for the Mark as Read/Unread action
-     */
-    showContextMenu(
-        type,
-        event,
-        selection,
-        contextMenuAnchor,
-        reportID,
-        reportAction,
-        draftMessage,
-        onShow = () => {},
-        onHide = () => {},
-        isArchivedRoom = false,
-        isChronosReport = false,
-        isPinnedChat = false,
-        isUnreadChat = false,
-    ) {
-        const nativeEvent = event.nativeEvent || {};
-        this.contextMenuAnchor = contextMenuAnchor;
-        this.contextMenuTargetNode = nativeEvent.target;
-
-        // Singleton behaviour of ContextMenu creates race conditions when user requests multiple contextMenus.
-        // But it is possible that every new request registers new callbacks thus instanceID is used to corelate those callbacks
-        this.instanceID = Math.random().toString(36).substr(2, 5);
-
-        this.onPopoverShow = onShow;
-        this.onPopoverHide = onHide;
-
-        this.getContextMenuMeasuredLocation().then(({x, y}) => {
-            this.setState({
-                cursorRelativePosition: {
-                    horizontal: nativeEvent.pageX - x,
-                    vertical: nativeEvent.pageY - y,
-                },
-                popoverAnchorPosition: {
-                    horizontal: nativeEvent.pageX,
-                    vertical: nativeEvent.pageY,
-                },
-                type,
-                reportID,
-                reportAction,
-                selection,
-                isPopoverVisible: true,
-                reportActionDraftMessage: draftMessage,
-                isArchivedRoom,
-                isChronosReport,
-                isPinnedChat,
-                isUnreadChat,
-            });
-        });
+   * Show the ReportActionContextMenu modal popover.
+   *
+   * @param {string} type - context menu type [EMAIL, LINK, REPORT_ACTION]
+   * @param {Object} [event] - A press event.
+   * @param {String} [selection] - Copied content.
+   * @param {Element} contextMenuAnchor - popoverAnchor
+   * @param {String} reportID - Active Report Id
+   * @param {Object} reportAction - ReportAction for ContextMenu
+   * @param {String} draftMessage - ReportAction Draftmessage
+   * @param {Function} [onShow] - Run a callback when Menu is shown
+   * @param {Function} [onHide] - Run a callback when Menu is hidden
+   * @param {Boolean} isArchivedRoom - Whether the provided report is an archived room
+   * @param {Boolean} isChronosReport - Flag to check if the chat participant is Chronos
+   * @param {Boolean} isPinnedChat - Flag to check if the chat is pinned in the LHN. Used for the Pin/Unpin action
+   * @param {Boolean} isUnreadChat - Flag to check if the chat is unread in the LHN. Used for the Mark as Read/Unread action
+   */
+  const showContextMenu = (
```

Comment:
I believe using `useImperativeHandle` is more suitable for these cases where we're calling functions from a component, but I know we have an uncommon setup here, so if it's too much trouble it's all good 👍🏻